adapters/satori/src/sdk/subgraphDetailsV2.ts

```diff
@@ -0,0 +1,50 @@
+import { ASSET, SYMBOL, SUBGRAPH_URL } from "./config";
+
+export interface OutputDataSchemaRow {
+    block_number:number
+    timestamp:number
+    user_address:string
+    token_address:string
+    token_symbol:string
+    token_balance:number
+}
+
+
+export const getUserTVLByBlock = async (
+    blockNumber: number,
+    timestamp: number,
+):Promise<OutputDataSchemaRow[]> =>  {
+    let subgraphUrl = SUBGRAPH_URL;
+    let blockQuery = blockNumber !== 0 ? `block: {number: ${blockNumber}}` : ``;
+    let skip = 0;
+    let result: OutputDataSchemaRow[] = [];
```

Comment:
@miler012 do you need to put this in a while loop to make sure all records related to a block number are fetched.